doc/classes/ShapeCast2D.xml

```diff
@@ -0,0 +1,182 @@
+		This node can detect multiple collision objects. If you want to use this just like [RayCast2D], use corresponding [code]get_closest_*[/code] methods instead.
+		[b]Note:[/b] shape casting is more computationally expensive compared to ray casting.
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="add_exception">
+			<return type="void" />
+			<argument index="0" name="node" type="Object" />
+			<description>
+				Adds a collision exception so the shape does not report collisions with the specified node.
+			</description>
+		</method>
+		<method name="add_exception_rid">
+			<return type="void" />
+			<argument index="0" name="rid" type="RID" />
+			<description>
+				Adds a collision exception so the shape does not report collisions with the specified [RID].
+			</description>
+		</method>
+		<method name="clear_exceptions">
+			<return type="void" />
+			<description>
+				Removes all collision exceptions for this shape.
+			</description>
+		</method>
+		<method name="force_shapecast_update">
+			<return type="void" />
+			<description>
+				Updates the collision information for the shape. Use this method to update the collision information immediately instead of waiting for the next [code]_physics_process[/code] call, for example if the shape or its parent has changed state.
+				[b]Note:[/b] [code]enabled == true[/code] is not required for this to work.
+			</description>
+		</method>
+		<method name="get_closest_collider" qualifiers="const">
+			<return type="Object" />
+			<description>
+				Returns the first object that the shape intersects, or [code]null[/code] if no object is intersecting the shape (i.e. [method is_colliding] returns [code]false[/code]).
+			</description>
+		</method>
+		<method name="get_closest_collider_shape" qualifiers="const">
+			<return type="int" />
+			<description>
+				Returns the shape ID of the first object that the shape intersects, or [code]0[/code] if no object is intersecting the shape (i.e. [method is_colliding] returns [code]false[/code]).
+			</description>
+		</method>
+		<method name="get_closest_collision_normal" qualifiers="const">
+			<return type="Vector2" />
+			<description>
+				Returns the normal of the closest intersecting object's shape at the collision point.
+			</description>
+		</method>
+		<method name="get_closest_collision_point" qualifiers="const">
+			<return type="Vector2" />
+			<description>
+				Returns the collision point at which the shape intersects the closest object.
+				[b]Note:[/b] this point is in the [b]global[/b] coordinate system.
+			</description>
+		</method>
+		<method name="get_closest_collision_safe_distance" qualifiers="const">
```

Comment:
The physics server actually returns [0..1] fractions, not distances, so either these functions should be renamed or the distance needs to be calculated from the motion length.

Reply:
`get_closest_collision_safe_fraction` or `get_closest_collision_safe_motion_fraction()` then? Kind of verbose though.

Reply:
It should be either `get_closest_collision_safe_fraction()` and a clearer explanation that it gives fractions between 0 and 1, or `get_closest_collision_safe_distance()` and in this case the method needs to be updated to return `motion_length * fraction`.